### DISCUSSION TITLE
Getting PING response with JedisPooled

### DISCUSSION THREAD

**Discussion by dmcauslan** [xetorthio/jedis#3003] [Q&A] [answered]
I use the result of PING as part of an app's health endpoint.  

With JedisPooled it looks like the only way to call PING is to use a protocol command, and the result seems to be an anonymous class. Is there a way to determine whether the command was successful or failed, and to get the "HELLO" response back?

Kotlin 1.6.21
Java 17.0.2
Jedis 4.2.3
Redis 6.2.5

```kotlin
import redis.clients.jedis.JedisPooled
import redis.clients.jedis.Protocol

val jedis = JedisPooled("localhost", 6379)
val result = jedis.sendCommand(Protocol.Command.PING, "HELLO")
result.javaClass

>>> class [B
```

**Comment by sazzad16** [comment, marked as answer]
`SafeEncoder.encode((byte[]) result)` should give you the "HELLO".

---

Besides that, you can provide `PooledConnectionProvider` with which would have more control. Here is the Java code:

```
PooledConnectionProvider provider = new PooledConnectionProvider(...);
JedisPooled jedis = new JedisPooled(provider);
try (Connection connection = provider.getConnection()) {
  connection.ping();
}
```

**Reply by dmcauslan** [reply]
@sazzad16 That worked, thanks!

Did I miss that in the docs somewhere?

```kotlin
import redis.clients.jedis.JedisPooled
import redis.clients.jedis.Protocol
import redis.clients.jedis.util.SafeEncoder

val jedis = JedisPooled("localhost", 6379)
val result = jedis.sendCommand(Protocol.Command.PING, "HELLO")
val stringResult = SafeEncoder.encode(result as ByteArray)
stringResult

>>> HELLO
```
